### ISSUE TITLE
Task:  bootstrap - persist metadata files in the end

### ISSUE THREAD

**Issue by MVrachev** [vmware/repository-service-tuf-worker#248]
### What is the task about?

While working on issue https://github.com/vmware/repository-service-tuf-api/issues/181 I noticed that if there is an exception in https://github.com/vmware/repository-service-tuf-worker/blob/5ac572a5d05558d915f3576e770490bc2e8bbd89/repository_service_tuf_worker/repository.py#L395 in the end of the function then the task is considered as failed, but we have already saved all metadata files.
I suggest we move the persisting of metadata files just before the end before creating the final response.

This will make sure that the next time we try to do bootstrap we can save the new files without any errors.

Also, it has the side benefit that it removes the confusion from the user because if a user sees that the bootstrap has failed and we successfully remove the lock, meaning the `api/v1/get/bootstrap` says the system is ready for bootstrap, but there are already existing metadata files it could be confusing for the user.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by MVrachev** [comment]
I have a small unpolished implementation on this issue.
If somebody decides to work on it can write me.

**Comment by theycallmeaabie** [comment]
I would like to work on this. @MVrachev could you share your implementation? I'd like to use it as a starting point.

**Comment by MVrachev** [comment]
@theycallmeaabie check my comment I spoke about this:
https://github.com/repository-service-tuf/repository-service-tuf-api/issues/181#issuecomment-1501951203
I don't have an implementation for it.

Look at the code and let me know if anything is unclear.

**Comment by vickysharma-prog** [comment]
Hi @MVrachev,

I went through the bootstrap implementation and storage layer carefully to understand the full scope of this fix.

**Root cause (verified):**
In `_bootstrap_finalize`, metadata files are persisted in two stages:

1. `_bootstrap_online_roles()` persists targets, snapshot, timestamp and delegated roles incrementally
2. `self._persist(root, Root.type)` persists root metadata

If any exception occurs between these steps, partial files are already written to storage but
`write_repository_settings("BOOTSTRAP", 
task_id)` is never called - leaving the system locked and inconsistent.

**Storage layer gap (verified):**
`IStorage` interface currently has no `delete()` method. To safely clean up partial files on failure, we need:
- `delete()` abstract method in `IStorage`
- Implementation in `LocalStorage`
- Implementation in `AWSS3`

**Proposed fix:**
Wrap persist operations in try/except:
- Track all persisted filenames
- On exception: delete persisted files release bootstrap lock, rethrow error

This ensures a failed bootstrap leaves the system clean and ready for retry without manual intervention.

I have started working on this and will open a PR soon!

@MVrachev is there anything else I should consider before opening the PR?

**Comment by theycallmeaabie** [comment]
Hey @MVrachev, sorry for the delay, took a bit to work through the right approach. Went with the robust fix 
(tracking and rolling back partial files on failure) rather than just deferring persists, which I think handles the failure cases 
more completely.                                                                                                                    
@vickysharma-prog if you're still interested in this one, happy to get your eyes on the PR.

**Comment by MVrachev** [comment]
You are right @vickysharma-prog with this suggestion, I see a pr from @theycallmeaabie.
Please, @theycallmeaabie, before starting to work on something, at least comment on the issue to try to get a response from @vickysharma-prog if he will continue his work.

I don't want you both working on the same thing and for one of you to feel disappointed.
